packages/sdk/src/marketplaces/Opensea.ts

```diff
@@ -48,12 +67,100 @@ export class Opensea implements Marketplace<Order> {
       networkName: Opensea.getNetwork(chainId),
     });
     this.address = address;
+    this.provider = provider;
   }
 
   static supportsChainId(chainId: number): boolean {
     return openseaSupportedChainIds.includes(chainId);
   }
 
+  async approveTakeOrderAsset(order: Order): Promise<void> {
+    const {
+      protocolData: { parameters },
+    } = order;
+
+    const { consideration } = parameters;
```

Comment:
I would suggest switching to `makerAssetBundle` and `takerAssetBundle` and determining which one you need to approve based on `side` param.

Reply:
No longer applicable since it's not using parsing assets; also, as mentioned before, `makerAssetBundle` and `takerAssetBundle` are vestiges of legacy wyvern js stuff and shouldn't be relied upon.